app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,60 @@
+class OrdersController < ApplicationController
+  include OrdersHelper
+
+  before_action :authenticate_user!, only: :add_to_cart
+
+  def authenticate_user!
+    unless current_user
+      redirect_to login_path, alert: 'You need to log in to add products to the cart'
+    end
+  end
+
+  def add_to_cart
+    filter_params = {
```

Comment:
I believe the same result could be achieve by using `params.to_h`. An even cleaner approach would be to also use `params.permit(...)` so that we can select the params we want to allow